### PR TITLE
feat: add autostart feature

### PR DESCRIPTION
**Pull request by thibaultboursier**
👉 The idea to add autostart comes from [this issue](https://github.com/thibaultboursier/use-timer/issues/47).

If consumer provides `shouldAutostart` parameter with `true` value, the timer will start automatically.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ export type Config = {
   interval: number;
   onTimeOver?: () => void;
   onTimeUpdate?: (time: number) => void;
+  shouldAutostart: boolean;
```
Comment (dancheskus) [inline src/types.ts:11]: isn't `autostart` better than `shouldAutostart`?

File: README.md [hunk 2]
```diff
@@ -109,12 +109,13 @@ const { time, start, pause, reset, status } = useTimer({
...
-| interval     | number   | 1000          | The interval in milliseconds                                                           |
-| onTimeOver   | function |               | Callback function that is called when time is over                                     |
-| onTimeUpdate | function |               | Callback function that is called when time is updated                                  |
-| step         | number   | 1             | the value to add to each increment / decrement                                         |
-| timerType    | string   | "INCREMENTAL" | the choice between a value that increases ("INCREMENTAL") or decreases ("DECREMENTAL") |
+| Property        | Type     | Default value | Description                                                                            |
+| --------------- | -------- | ------------- | -------------------------------------------------------------------------------------- |
+| endTime         | number   | null          | The value for which timer stops                                                        |
```
Comment (dancheskus) [inline README.md:114]: I think `endTime` should be `null` by default in `INCREMENTAL` mode and `0` in `DECREMENTAL` mode

File: src/types.ts (hunk 1 above)
Comment (thibaultboursier) [reply src/types.ts:11]: I prefer use `has`, `should`, `is`, ... prefixes for boolean variables. Why do you think it's better to name it `autostart`?

File: README.md (hunk 2 above)
Comment (thibaultboursier) [reply README.md:114]: Why not... Maybe we could check this point in an other PR, as it's not related :-)

File: src/types.ts (hunk 1 above)
Comment (dancheskus) [reply src/types.ts:11]: I think `autostart: true` is describable enough, but it is up to you.

Comment (thibaultboursier) [reply src/types.ts:11]: I'll make some concessions ;-) It's sometimes a hard task to define some variable names... Let's use `autostart`, it's shorter.

Comment (thibaultboursier) [reply src/types.ts:11]: i'll make the change soon.

**Pull request merged**